### ISSUE TITLE
gatsby-source-drupal - Skip certain JSONAPI resources (file--file)

### ISSUE THREAD

**Issue by j-lasak** [gatsbyjs/gatsby#13664]
The Drupal backend I'm using has many images that won't be used in the Gatsby app, but are downloaded regardless due to the way gatsby-source-drupal gatsby image processing works. This is causing a longer build time than I'd like.
 
I think it's possible to include the image data with the node endpoints via an "include" using the filter option. Something like:

`filters: {`
`          // collection : filter`
`          'node/node_type': "include=field_image",    `    
`},`

If this works, is there a way to skip certain JSONAPI resources? I need the 'file--file' resource enabled on Drupal due to the "include" above, but don't want gatsby-source-drupal to fetch every single image.

If anyone has any ideas about this or limiting image downloads in general, I'd appreciate it.

Thanks!

**Comment by j-lasak** [comment]
Ah, I see it's possible to add a filter where no image resources are pulled in with something like

`'file--file': 'filter[filemime]=none'`

But now that I'm using the "include" as mentioned above, I see that data isn't getting imported to graphQL.

Anyone have any ideas about importing only selected images?

**Comment by freiksenet** [comment]
Thanks for this PR! We really appreciate you taking the time to open it.

In the interest of moving this along most effectively, we strongly encourage our contributors to provide an example project which we can then use to validate the changes in this PR. This will allow us to quickly and easily ensure that this does indeed address the purported feature or fix in this PR!

An example project:

- can be run locally or credentials have been provided (so ideally your Drupal instance or a copy)
    - Feel free to message on Twitter, Discord, etc.
- can be used to validate this functionality
   - In other words, the necessary functionality has been enabled/disabled in the project for us to easily validate these > changes>

Thanks for this PR and thanks for using Gatsby!

:purple_heart:

**Comment by rcaracaus** [comment]
A lot of our Drupal/Gatsby sites have quite a lot of images that take a long time to download from the Drupal server, I have had to delete this section from gatsby-source-drupal so our developers can work on the site without needing to download all the files. We use Cloudinary for image transformations.

![Screen Shot 2019-08-09 at 2 17 10 PM](https://user-images.githubusercontent.com/1032160/62803743-db6ee880-bab0-11e9-87b5-d666104fb2d5.png)

Maybe we should reopen this ticket or add an option to prevent file download.
